### PR TITLE
Add LGTM.com configuration file

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Add a configuration file for the LGTM.com static code analysis tool. It requires no CI integration, as the LGTM.com tool continuously scans the python-blosc GitHub repository for security and/or code vulnerabilities. You can find the alerts here:
https://lgtm.com/projects/g/Blosc/python-blosc/

The YAML configuration file for the LGTM static analysis tool can be either `lgtm.yml` or `.lgtm.yml`:
            https://lgtm.com/help/lgtm/lgtm.yml-configuration-file

### CODE REVIEW

File: .lgtm.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# The python-blosc GitHub repository is continuously scanned by the LGTM.com
+# tool for security and/or code vulnerabilities. You can find the alerts here:
+# https://lgtm.com/projects/g/Blosc/python-blosc/
+
+path_classifiers:
+  generated:
+    # suppress alerts in vendored files
+    - cpuinfo.py
+    - blosc/_version.py
+    - blosc/c-blosc
```
Comment (FrancescAlted) [inline .lgtm.yml:12]: That means that c-blosc alerts should be suppressed, but I am seeing several alerts in https://lgtm.com/projects/g/Blosc/python-blosc/.  Am I missing something?

Comment (DimitriPapadopoulos) [reply .lgtm.yml:12]: LGTM.com scans the default branch of repositories
- I expect alerts related to vendored code to disappear when this file is merged in the default branch `main`.
- Other alerts may be relevant. You can hose to fix them or ignore later on.

**Comment by FrancescAlted** [comment]
LGTM

**Comment by DimitriPapadopoulos** [comment]
Because Semmle has joined GitHub, LGTM.com will be deprecated and replaced by GitHub code scanning.

[The next step for LGTM.com: GitHub code scanning!](https://github.blog/2022-08-15-the-next-step-for-lgtm-com-github-code-scanning/)

As far as I can understand, in simple cases such as this one, automated pull requests will be created to help us migrate:
> We will do our best to help migrate repositories that actively use LGTM.com to flag potential security issues in their pull requests. For those repositories, we will create pull requests that add a [GitHub Actions workflow that runs code scanning](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/setting-up-code-scanning-for-a-repository#setting-up-code-scanning-manually).

**Pull request merged**